clap_complete/tests/snapshots/basic.fish

```diff
@@ -1,10 +1,36 @@
-complete -c my-app -n "__fish_use_subcommand" -s c
-complete -c my-app -n "__fish_use_subcommand" -s v
-complete -c my-app -n "__fish_use_subcommand" -s h -l help -d 'Print help'
-complete -c my-app -n "__fish_use_subcommand" -f -a "test" -d 'Subcommand with a second line'
-complete -c my-app -n "__fish_use_subcommand" -f -a "help" -d 'Print this message or the help of the given subcommand(s)'
-complete -c my-app -n "__fish_seen_subcommand_from test" -s d
-complete -c my-app -n "__fish_seen_subcommand_from test" -s c
-complete -c my-app -n "__fish_seen_subcommand_from test" -s h -l help -d 'Print help'
-complete -c my-app -n "__fish_seen_subcommand_from help; and not __fish_seen_subcommand_from test help" -f -a "test" -d 'Subcommand with a second line'
-complete -c my-app -n "__fish_seen_subcommand_from help; and not __fish_seen_subcommand_from test help" -f -a "help" -d 'Print this message or the help of the given subcommand(s)'
+# Print an optspec for argparse to handle cmd's options that are independent of any subcommand.
```

Comment:
Every commit should pass tests.  Please squash this into the previous commit

Reply:
I was afraid of merge conflict when rebasing on refactoring. I'll squash into one when ready.